src/components/pages/GenerationPage/PetitionListItem.js

```diff
@@ -1,24 +1,25 @@
-import React, { useContext } from 'react';
+import React, { useState } from 'react';
 import { PetitionListItemStyled, PetitionCellStyled } from './PetitionListItem.styled';
-import { GenerationContext } from './GenerationPage';
 import GeneratePetitionModal from './GeneratePetitionModal/GeneratePetitionModal';
 
-function PetitionListItem({ petition }) {
-  const { showGenerationModal, setShowGenerationModal, handlePetitionSelect } = useContext(
-    GenerationContext
-  );
-
+function PetitionListItem({ petition, handlePetitionSelect }) {
+  const [isVisible, setVisible] = useState(false);
+  const handleClick = () => {
+    handlePetitionSelect(petition);
+    setVisible(true);
```

Comment:
This always opens a modal for me, even if there are empty required fields. I didn't see an easy fix now that ``setVisible`` is defined here though.

Reply:
Thanks for catching that, my first iteration of the fix handled that edge case but my current iteration doesn't. I have a fix in mind that I'll push after some testing.